Purchases/Public/RCPurchases.m

```diff
@@ -784,6 +790,7 @@ - (void)collectDeviceIdentifiers {
 
 - (void)applicationDidBecomeActive:(__unused NSNotification *)notif {
     [self updateAllCachesIfNeeded];
+    [self syncSubscriberAttributesIfNeeded];
```

Comment:
this guarantees order of operations between these two, although if the caches calls fail, we'll still try to sync subscriber attributes. To fix that would be more complex, and we can take care of it, but it felt a bit out of scope since this will already fix the majority of cases.